src/hypothesis/stateful.py

```diff
@@ -518,3 +575,8 @@ def execute_step(self, step):
             )
             for target in rule.targets:
                 self.bundle(target).append(VarReference(name))
+
+        for invar in self.invariants():
+            if invar.precondition is not None and not invar.precondition(self):
```

Comment:
I needed to duplicate this line, and ran up against live length limits, so I changed all three checks to just use truthiness.